Explorer/Assets/DCL/Billboard/System/BillboardSystem.cs

```diff
@@ -0,0 +1,64 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using DCL.Billboard.Extensions;
+using DCL.CharacterCamera;
+using DCL.ECSComponents;
+using ECS.Abstract;
+using ECS.Groups;
+using ECS.Unity.Transforms.Components;
+using ECS.Unity.Transforms.Systems;
+using UnityEngine;
+
+namespace DCL.Billboard.System
+{
+    [UpdateInGroup(typeof(SyncedSimulationSystemGroup))]
+    [UpdateAfter(typeof(UpdateTransformSystem))]
+    public partial class BillboardSystem : BaseUnityLoopSystem
+    {
+        private readonly IExposedCameraData exposedCameraData;
+
+        public BillboardSystem(World world, IExposedCameraData exposedCameraData) : base(world)
+        {
+            this.exposedCameraData = exposedCameraData;
+        }
+
+        protected override void Update(float t)
+        {
+            Quaternion cameraRotationAxisZ = Quaternion.Euler(0, 0, exposedCameraData.WorldRotation.eulerAngles.z);
+            Vector3 cameraPosition = exposedCameraData.WorldPosition;
+            UpdateRotationQuery(World, cameraPosition, cameraRotationAxisZ);
+        }
+
+        [Query]
+        private void UpdateRotation(
+            [Data] in Vector3 cameraPosition,
+            [Data] in Quaternion cameraRotationAxisZ,
+            ref TransformComponent transform,
+            in PBBillboard billboard
+        )
+        {
+            if (billboard.BillboardMode is BillboardMode.BmNone)
+                return;
+
+            Vector3 forward = transform.Transform.forward;
+
+            if (billboard.UseX() || billboard.UseY())
+            {
+                forward = cameraPosition - transform.Cached.WorldPosition;
+
+                if ((billboard.BillboardMode & BillboardMode.BmY) == 0) forward.x = 0;
+                if ((billboard.BillboardMode & BillboardMode.BmX) == 0) forward.y = 0;
+
+                forward.Normalize();
+            }
+
+            Quaternion rotation = forward != Vector3.zero ? Quaternion.LookRotation(forward) : Quaternion.identity;
+
+            if (billboard.UseZ())
+                rotation *= cameraRotationAxisZ;
+
+            transform.Transform.rotation = rotation;
```

Comment:
We should update the cached rotation value as well as other systems may consume it

Reply:
Ok, I added an Apply method to the TransformComponent to save SRP

Reply:
It runs about 0.15 ms for 200 entities and 0.35 ms for 500 entities

Specs: 
- Apple M1 Pro
- 32 GB

Reply:
The performance looks good to me, thanks. So at this point let's assume we don't need to optimize it further (despite we know it's doable with Jobs)